src/test/c_api_stress_tests.cpp

```diff
@@ -0,0 +1,441 @@
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::retireSpecificVersionUsed,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, AddNewVersionDuringGetMetadataLoad) {
+    bool performWholeConfigReload = false;                        // we just need to have all model versions rechecked
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::defaultVersionAdd,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RemoveDefaultVersionDuringGetMetadataLoad) {
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::MODEL_NAME_MISSING};  // we hit when all config changes finish to propagate
+    std::set<StatusCode> allowedLoadResults = {};
+    // we need whole config reload since there is no other way to dispose
+    // all model versions different than removing model from config
+    bool performWholeConfigReload = true;
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::defaultVersionRemove,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, ChangeToShapeAutoDuringGetMetadataLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_NAME_MISSING};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::changeToAutoShape,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RemovePipelineDefinitionDuringGetMetadataLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::MODEL_NAME_MISSING};  // when pipeline is retired
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::removePipelineDefinition,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, ChangedPipelineConnectionNameDuringGetMetadataLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_NAME_MISSING};
```

Comment:
StatusCode::PIPELINE_DEFINITION_NOT_LOADED_ANYMORE instead of MODEL_NAME MISSING?

Reply:
and again as required?